### PR TITLE
[FrameworkBundle] Added --sort option for TranslationUpdateCommand

### PR DESCRIPTION
**Pull request by k0d3r1s**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #33014 
| License       | MIT
| Doc PR        | symfony/symfony-docs#12136
Added sorting option for translation:update command

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 1]
```diff
@@ -260,6 +265,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+                    $sort = strtolower($sort);
+                    if (!\in_array($sort, self::SORT_ORDERS, true)) {
+                        $errorIo->error(['Wrong sort order', 'Supported formats are: '.implode(', ', self::SORT_ORDERS).'.']);
+
+                        return 1;
+                    }
+
+                    if (self::DESC === $sort) {
+                        rsort($list);
+                    } else {
```
Comment (Simperfit) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: is the else really needed ? If the sort option is wrong it should not come here, I think we can drop the else.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php [hunk 2]
```diff
@@ -32,6 +32,14 @@ public function testDumpMessagesAndClean()
         $this->assertRegExp('/1 message was successfully extracted/', $tester->getDisplay());
     }
 
+    public function testDumpSortedMessagesAndClean()
```
Comment (Simperfit) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php]: can you please test with a wrong sort ?

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 3]
```diff
@@ -88,7 +93,7 @@ protected function configure()
 message.
 
 Example running against a Bundle (AcmeBundle)
-  <info>php %command.full_name% --dump-messages en AcmeBundle</info>
+  <info>php %command.full_name% --dump-messages --sort=asc en AcmeBundle</info>
```
Comment (Simperfit) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: I think you can add another example to explain the sort option.

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 1 above)
Comment (k0d3r1s) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: how come? sort for asc order and rsort for reverse. or i misunderstood your comment?

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php]: and also with `--sort=desc` please.

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: The `else` looks correct and necessary to me.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php (hunk 2 above)
Comment (k0d3r1s) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php]: done

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 3 above)
Comment (k0d3r1s) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: two are not enough?

**Comment by k0d3r1s** [comment]
what happened to travis builds? I just applied patch and everything went wrong

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 4]
```diff
@@ -36,6 +36,10 @@
  */
 class TranslationUpdateCommand extends Command
 {
+    protected const ASC = 'asc';
+    protected const DESC = 'desc';
+    protected const SORT_ORDERS = [self::ASC, self::DESC];
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: `const`s should be `private` (the class is marked as `final` anyway).

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 5]
```diff
@@ -88,11 +93,11 @@ protected function configure()
 message.
 
 Example running against a Bundle (AcmeBundle)
-  <info>php %command.full_name% --dump-messages en AcmeBundle</info>
+  <info>php %command.full_name% --dump-messages --sort=asc en AcmeBundle</info>
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: I would remove the `--sort` in the 2 existing commands and add a sentence explaining the `sort` option with a new example.

**Review by fabpot — APPROVED** [review]
With some small nitpicks.

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 4 above)
Comment (k0d3r1s) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @k0d3r1s.

**Pull request merged**